### DISCUSSION TITLE
installazione di ZigBee2MQTT su Raspberry PI zero W non riuscita

### DISCUSSION THREAD

**Discussion by PiuCurioso** [indomus/forum#694] [Networking]
Carissimi buongiorno vi scrivo per un aiuto sulla seguente questione: sono un possessore di Raspberry Pi Zero W e sto tentando di creare una rete Zigbee per gestire alcuni dispositivi IoT.Ho installato sul Raspi PI Zero W il broker MQTT Mosquitto e ZigBee2MQTT seguendo la guida su indomus []https://indomus.it/guide/come-installare-e-configurare-zigbee2mqtt-su-raspbian-di-raspberry-pi/)   il problema è proprio ZigBee2MQTT la cui installazione non va a buon fine sul mio Raspberry Pi zero W per cui non mi permette di utilizzare l'interfaccia web di ZigBee2MQTT. Spero mi possiate dare una mano e vi ringrazio tutti per quello che fate e che ho appreso grazie al sito di indomus.Buon pomeriggio
Ciao Ciao

**Comment by snakuzzo** [comment]
Senza dettagli è impossibile darti una mano.
Posso solo dirti che Z2M sul PiZero W funziona

**Reply by riccionese** [reply]
Anche io uso un Pi ZeroW ma con solo Z2M e senza MQTT (punto ad un altro Pi 4 con MQTT e HA) e confermo che non ho problemi né con la funzionalità in sé né con l'interfaccia WEB. 
Così a naso, non credo che il problema possa essere la convivenza con Mosquitto, che lavora su altre porte ma come dice Diego, senza dettagli...

**Comment by PiuCurioso** [comment]
Grazie carissimi adesso sono fuori casa, non appena posso rifaccio l'installazione di ZigBee2MQTT e vi posto i log.Grazie mille
Un caro saluto

**Comment by PiuCurioso** [comment]
![image](https://user-images.githubusercontent.com/55890118/132399215-ba3a63fe-2319-4487-84dd-64378777ffc0.png)
Ahimè l'installazione di ZigBee2MQTT sembra andata a buon fine ma non mi riesco a collegare all'interfaccia web di ZigBeeMQTT :( Grazie mille per il vostro aiuto notte

**Comment by PiuCurioso** [comment]
**SCHERMATE RIMOSSE**

Carissimi vi ho postato i log di ZigBee2MQTT grazie per il vostro aiuto
vi auguro una buona giornata

**Reply by snakuzzo** [reply]
1. Se hai una CC2530 o CC2531 con l'ultima versione di Zigbee2mqtt (1.21) ci sono dei problemi. Prova a passare alla 1.20
2. Non puoi assolutamente postare le schermate del client che usi con tutti i log. Quindi ora rimuovo tutto
3. Se proprio vuoi postare i log, fallo usando il markdown
4. Se continui a postare in questo modo saremo costretti a bannarti (non è la prima volta che ti richiamiamo)

**Reply by NdR91** [reply]
Scusa ma chi pretendi che si metta a guardare 100 screenshots buttati a caso???

Semmai, ma proprio se è necessario far vedere tutte queste righe di log, lo copi e lo incolli su pastebin.com e poi poni la domanda che devi fare...

**Reply by PiuCurioso** [reply]
Carissimi chiedo scusa ma non avendo dimestichezza con le regole del forum ho sbagliato e di questo chiedo scusa :( Mi dispiace per cui Snakuzzo rimuovi pure.Cosa dovrei postar per cercare di risolvere il problema solo i log di ZigBee2MQTT ?e in che modo?Non ho capito cos'è il markdown.L'antenna che uso e questa descritta in questo articolo https://indomus.it/progetti/adottare-ser2net-ed-esp-01-su-z-bee-duo-per-gestione-remota-con-zigbee2mqtt/
Grazie mille per il vostro aiuto e mi scuso per aver postato male le immagini
Buona giornata

**Reply by snakuzzo** [reply]
Leggi il regolamento [QUI](https://github.com/indomus/forum/blob/main/README.md)...c'è scritto come postare. Se non sai cosa sia il markdown, ti basta una ricerca.
Se proprio non riesci ad usare il markdown (ma mi pare troppo strano visto che si tratta di fare 1 click), come già ti è stato suggerito, puoi usare pastebin.com. Ma schermate di log proprio no !!! E copia solo un estratto del log ... non 1020012301203120 righe inutili

**Reply by PiuCurioso** [reply]
> Leggi il regolamento [QUI](https://github.com/indomus/forum/blob/main/README.md)...c'è scritto come postare. Se non sai cosa sia il markdown, ti basta una ricerca.
> Se proprio non riesci ad usare il markdown (ma mi pare troppo strano visto che si tratta di fare 1 click), come già ti è stato suggerito, puoi usare pastebin.com. Ma schermate di log proprio no !!! E copia solo un estratto del log ... non 1020012301203120 righe inutili

Ok grazie mille Snakuzzo

**Reply by PiuCurioso** [reply]
>     1. Se hai una CC2530 o CC2531 con l'ultima versione di Zigbee2mqtt (1.21) ci sono dei problemi. Prova a passare alla 1.20
> 
>     2. Non puoi assolutamente postare le schermate del client che usi con tutti i log. Quindi ora rimuovo tutto
> 
>     3. Se proprio vuoi postare i log, fallo usando il markdown
> 
>     4. Se continui a postare in questo modo saremo costretti a bannarti (non è la prima volta che ti richiamiamo)

Caro Snakuzzo la mia antenna è questa https://indomus.it/progetti/adottare-ser2net-ed-esp-01-su-z-bee-duo-per-gestione-remota-con-zigbee2mqtt/
operativamente che comandi dovrei utilizzare per passare alla versione 1.20?
Grazie mille
Buona giornata

**Reply by snakuzzo** [reply]
Se usi quella non c'è bisogno di fare downgrade. Quando ti ho scritto del downgrade non ero a conoscenza del chip in tuo possesso.
Tu stesso dovresti sapere il chip che hai sulla chiavetta, visto che ce l'hai e che avrai letto l'articolo.

**Reply by PiuCurioso** [reply]
Ok grazie Snakuzzo allora ti posto in modo corretto il log utilizzando la funzione che mi hai indicato(mi informo come si fa).Quando dici che devo sintitetizzare il log significa che da quando do il comando di visualizzare il log devo mandarti quante righe di log? Io ne ho mandate tante in quanto non so cosa sia importante per capire un eventuale problema Mi puoi indicare a cosa devo prestare attenzione nell'estrarre i dati importanti dal log.Grazie mille ciao

**Reply by snakuzzo** [reply]
Non posso dirtelo io...devi riuscire a capirlo tu leggendo il log. Vedi più o meno quando cominci ad avere l'errore. posti l'errore e fine.
Da quello che ho visto al volo dalle schermate orrende di prima, semplicemente non riesci ad aprire la seriale.
Ed a questo punto ti chiederei da subito che configurazione hai, perché mi sembra proprio un problema legato alla prima configurazione.
Ad ogni modo, come immagino ti avrà già indicato @gio-dot, [QUI](https://github.com/Gio-dot/Z-Bee-Duo) trovi tutte le informazioni per la configurazione

**Reply by PiuCurioso** [reply]
Ok Snakuzzo ti invio solo l'errore.Che significa quando dici" semplicemente non riesci ad aprire la seriale".?
"Ed a questo punto ti chiederei da subito che configurazione hai(come faccio a mandarti la configurazione?che dato devo copiare?), perché mi sembra proprio un problema legato alla prima configurazione(di cosa?della antenna?)."
Ti riferisci alla configurazione dell'antenna?Ti posso mandare qualche dato nello specifico?

**Reply by gio-dot** [reply]
Come ti dicevo di la, durante l'installazione non ci devono essere errori o warning particolari, quundi se provi ad installare, al primo errore che  vedi FERMATI, cerca di capirlo e se non lo capisci resta fermo e posta il log dove si vede l'errore usando magari Pastebin

**Reply by PiuCurioso** [reply]
> Come ti dicevo di la, durante l'installazione non ci devono essere errori o warning particolari, quundi se provi ad installare, al primo errore che vedi FERMATI, cerca di capirlo e se non lo capisci resta fermo e posta il log dove si vede l'errore usando magari Pastebin

Ok grazie mille Gio-dot
buona giornata a tutti

**Reply by PiuCurioso** [reply]
Carissimi, 
ho provato a reinstallare piu volte ZigBee2MQTT in una delle volte ho trovato questo errore, 

https://pastebin.com/raw/iCjbR6HS

ma non ne comprendo il significato.
Potrebbere essere questa la causa del fatto che a fine installazione non riesco a collegarmi all'interfaccia web di ZigBee2MQTT?
Grazie mille per il vostro aiuto
Buona serata

**Reply by snakuzzo** [reply]
Non hai capito l'errore che devi postare.
Con quello che hai postato ci facciamo poco

**Reply by PiuCurioso** [reply]
Grazie Snakuzzo devo provare ad installare nuovamente ZigBee2MQTT e postare gli errori.Grazie Notte

**Reply by PiuCurioso** [reply]
Caro Snakuzzo, ciao
ho googlato un pò per capire se altri hanno riscontrato problemi con il Raspberry PiZero W e ho trovato un articolo in cui a proposito della installazione di ZigBee2MQTT  riporta quanto segue:
La procedura standard è descritta qui:
https://www.zigbee2mqtt.io/getting_started/running_zigbee2mqtt.html
Ma su Raspberry PI Zero non funziona, perchè quel modo di installare node.js non va.
E suggerisce di: 
https://pastebin.com/raw/Nsi9xVSp
E conclude per la guida di configurazione rimando al sito ufficiale.

Può essere questa una soluzione? Che ne pensi? i comandi proposti sono affidabili? I repository utilizzati per scaricare node.js sono affidabili? Che ne pensi? o sono fuori strada e lascio stare questi suggerimenti?
Grazie mille e  ringrazio tutti per la  disponibilità 
Buona giornata

**Reply by snakuzzo** [reply]
Ho letto un po' quello che hai linkato, però da quello che vedo tu hai installato zigbee2mqtt e quindi hai anche installato nodejs.
Purtroppo non so dirti perché sul PiZeroW non ho fatto questo tipo di installazione.
Posso però dirti che qualche mese fa ho tirato su Zigbee2mqtt con Docker (e docker-compose) e non ho avuto alcun problema.
Trovi tutto documentato [QUI](https://www.zigbee2mqtt.io/information/docker.html) nel caso.

**Reply by PiuCurioso** [reply]
ok grazie mille Snakuzzo ma se io volessi seguire sul mio Raspberry PI Zero W l'installazione di ZigBee2MQTT con docker dovrei prima disinstallare ZigBee2MQTT e Node ( che ho gia installato?)( Mosquitto invece lo lascio stare?)e con quali comandi dovrei fare cio?
E poi potrei seguire la vostra guida
https://indomus.it/guide/come-installare-e-configurare-zigbee2mqtt-con-docker-su-raspbian-di-raspberry-pi/   ?
Grazie mille

**Reply by snakuzzo** [reply]
Non è che devi disinstallarlo per forza. L'importante è non andare in conflitto con l'altra installazione. Ti basta spegnere quel servizio e installare l'altro.
La guida che hai linkato dovrebbe andar bene.
Comunque per farlo devi prima studiare un attimino docker, non è che segui guide passo passo alla cieca. Se non sei pratico inutile andare a tentoni e al minimo problema chiedi qui...non è che possiamo seguirti passo passo h24.
Studia prima docker, vedi come funziona, sperimenta e poi fai le cose.

P.S. Mosquitto puoi anche lasciarlo così com'è...ma sappi che anche quello puoi metterlo su docker...abbiamo anche la guida

**Reply by PiuCurioso** [reply]
> Non è che devi disinstallarlo per forza. L'importante è non andare in conflitto con l'altra installazione. Ti basta spegnere quel servizio e installare l'altro.
> La guida che hai linkato dovrebbe andar bene.
> Comunque per farlo devi prima studiare un attimino docker, non è che segui guide passo passo alla cieca. Se non sei pratico inutile andare a tentoni e al minimo problema chiedi qui...non è che possiamo seguirti passo passo h24.
> Studia prima docker, vedi come funziona, sperimenta e poi fai le cose.
> 
> P.S. Mosquitto puoi anche lasciarlo così com'è...ma sappi che anche quello puoi metterlo su docker...abbiamo anche la guida

ok grazie mille Snakuzzo approfondirò Docker :)
ho trovato la guida per installare Mosquitto su Docker
 https://indomus.it/guide/come-installare-e-configurare-il-broker-mqtt-mosquitto-su-docker-su-raspbian/
grazie
ti auguro una buona giornata
Ciao Ciao

**Reply by PiuCurioso** [reply]
Caro Snakuzzo, ciao,

scusami se ti disturbo nuovamente, con l'intento di eseguire l'installazione di ZigBee2MQTT con Docker sul Raspberry PI Zero W, ho provato ad installare Docker utilizzando questa guida https://indomus.it/guide/come-installare-docker-su-raspberry-pi-con-raspbian/#verifiche
Ho seguito i comandi per l'installazione automatica, ma penso non sia andata a buon fine, per cui prima di tentare l'installazione manuale ( piu articolata) ti volevo mostrare l'esito della verifica. 
Mi scuso se non ho potuto usare il servizio pastebin( o il markdown) per incollare il testo della verifica in quanto ho provato ad estrarre il testo dallo screenshot che ho fatto  dal mio PC (collegato tramite SSH al Raspberry ) subito dopo aver eseguito il comando 
`docker run armhf/hello-world`
ma la funzione OCR di un applicazione web  mi ha restituito un testo incomprensibile, per cui ho preferito caricare lo screenshot su un sito on line e incollare su questo forum il relativo link https://i.postimg.cc/g2hh0hPp/Errore-per-indomus.jpg
Inoltre come consigliato ho installato anche il tool Docker Compose
Grazie mille per il tuo aiuto
Buon Sabato

**Reply by snakuzzo** [reply]
Devi eseguire anche questi step

```
sudo groupadd docker
sudo usermod -aG docker $USER
sudo reboot
```

**Reply by snakuzzo** [reply]
Ma io continuo a pensare che stai facendo questa installazione senza sapere effettivamente cosa stai facendo

**Reply by PiuCurioso** [reply]
Ti ringrazio caro Snakuzzo si e' vero non sono molto addentrato tecnicamente, ma quello che so e che installando docker sto creando un ambiente di virtualizzazione software in cui eseguire ZigBee2MQTT in modo totalmente isolato dal Sistema Operativo ( in questo caso Raspberry PI OS). Poi tecnicamente non so dirti di più e non ho ben chiaro quale sia la differenza tra la virtualizzazione hardware ( vedi Virtualbox)e la conteinerizzazione( che è una virtualizzazione software) Grazie mille PS se avrai modo di spiegarmelo in maniera non tecnica per i non addetti ai lavori, te ne sarei grato. 
Grazie ancora per avermi indicato i suddetti comandi. 
Ti auguro una buona domenica 
Ciao Ciao

**Reply by snakuzzo** [reply]
Non si tratta di spiegare in maniera non tecnica...è che certe cose necessitano di pratica e un po' di studio.
Capisco che tu voglia fare installazioni qua e là, ma non puoi andare avanti a copia-incolla purtroppo.
La nostra offerta di aiuto resta sempre valida comunque.
Prova ad attrezzarti...

Se poi non ti fidi e vuoi andare a chiedere in altri gruppi Telegram, basta dirlo...così chiudiamo questa discussione sul forum ;)

**Reply by PiuCurioso** [reply]
> Non si tratta di spiegare in maniera non tecnica...è che certe cose necessitano di pratica e un po' di studio.
> Capisco che tu voglia fare installazioni qua e là, ma non puoi andare avanti a copia-incolla purtroppo.
> La nostra offerta di aiuto resta sempre valida comunque.
> Prova ad attrezzarti...
> 
> Se poi non ti fidi e vuoi andare a chiedere in altri gruppi Telegram, basta dirlo...così chiudiamo questa discussione sul forum ;)

Ciao Snakuzzo, innanzitutto ringrazio sia te che Gio-dot per il vostro aiuto ed è proprio grazie a indomus( e quindi a voi) che ho imparato molte cose, pensa non sapevo assolutamente che esistessero dispositivi IoT ZigBee ( l'ho appreso grazie a indomus e Gio-dot mi ha illustrato ZigBee2MQTT + Mosquitto e il ZigBee Coordinator Z-Bee Duo con il modulo ESP-01( che ho acquistato ) cose per me tutte nuove che ho appreso grazie a voi. Ahimè la mia preparazione è abbastanza "povera" fatta leggendo i vari articoli su indomus e altri siti specializzati. Sto cercando di attrezzarmi cercando di trovare corsi validi, mi piacerebbe imparare Linux, mi picerebbe imparare a gestire una rete informatica con particolare riferimento alla configurazione di un firewall e delle VLAN. Ma trovare questi corsi gratuiti ( in quanto non devo fare come lavoro l'amministratore di rete -perchè ho già un lavoro-, l'informatica è un hobby) non è semplice. Ho cercato di contattare vari LUG ma quelli attivi sono solo al Nord ed io vivo al sud. Ma prima o poi riuscirò a frequentare un corso decente :) ecco perchè per il momento mi accontento di copiare e incollare ( come dicevi tu non ho le competenze per essere autonomo). Poi di voi mi fido eccome, ma se mi capita di chiedere "in giro" e solo perchè mi rendo conto, come dicevi tu stesso" che non posso chiedere a voi di seguirmi 24H su 24 in quanto avete i vostri impegni, che è cosa sacrosanta. Per cui ti prego di tenere "apertissimo" questo Topic  Grazie mille per il tempo che mi dedicate buona domenica

**Reply by PiuCurioso** [reply]
> Non posso dirtelo io...devi riuscire a capirlo tu leggendo il log. Vedi più o meno quando cominci ad avere l'errore. posti l'errore e fine.
> Da quello che ho visto al volo dalle schermate orrende di prima, semplicemente non riesci ad aprire la seriale.
> Ed a questo punto ti chiederei da subito che configurazione hai, perché mi sembra proprio un problema legato alla prima configurazione.
> Ad ogni modo, come immagino ti avrà già indicato @Gio-dot, [QUI](https://github.com/Gio-dot/Z-Bee-Duo) trovi tutte le informazioni per la configurazione

Grazie mille Snakuzzo per il tuo aiuto per l'aiuto di Gio-dot e di tutta indomus finalmente ho risolto il problema, adesso l'interfaccia web di ZigBee2MQTT funziona era un problema di configurazione.Grazie mille buona giornata
